api/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -225,6 +226,7 @@ private MethodDescriptor(
     this.type = Preconditions.checkNotNull(type, "type");
     this.fullMethodName = Preconditions.checkNotNull(fullMethodName, "fullMethodName");
     this.serviceName = extractFullServiceName(fullMethodName);
+    this.methodName = extractMethodName(fullMethodName);
```

Comment:
I don't think we should pre-compute this.

Reply:
Does this really matter? This should be a fairly cheap operation.

Reply:
If it's so cheap we shouldn't mind doing it every time. Really, I don't think we want the startup initialization overhead and memory usage for something that may never be used (consider Android for example).

Reply:
My concern is that then every caller will need to cache it, but alas. Done.